pkg/bootstrap/docker/up.go

```diff
@@ -0,0 +1,647 @@
+	server := c.OpenShiftHelper().Master(c.ServerIP)
+	return openshift.Login(initialUser, initialPassword, server, c.LocalConfigDir, c.originalFactory, c.command, out)
+}
+
+// CreateProject creates a new project for the current user
+func (c *ClientStartConfig) CreateProject(out io.Writer) error {
+	return openshift.CreateProject(initialProjectName, initialProjectDisplay, initialProjectDesc, "oc", out)
+
+}
+
+// ServerInfo displays server information after a successful start
+func (c *ClientStartConfig) ServerInfo(out io.Writer) error {
+	fmt.Fprintf(out, "OpenShift server started.\n"+
+		"The server is accessible via web console at:\n"+
+		"    %s\n\n"+
+		"You are logged in as:\n"+
+		"    User:     %s\n"+
+		"    Password: %s\n\n",
+		c.OpenShiftHelper().Master(c.ServerIP),
+		initialUser,
+		initialPassword)
+	return nil
+}
+
+// Factory returns a command factory that works with OpenShift server's admin credentials
+func (c *ClientStartConfig) Factory() (*clientcmd.Factory, error) {
+	if c.factory == nil {
+		cfg, err := kclientcmd.LoadFromFile(filepath.Join(c.LocalConfigDir, "master", "admin.kubeconfig"))
+		if err != nil {
+			return nil, err
+		}
+		defaultCfg := kclientcmd.NewDefaultClientConfig(*cfg, &kclientcmd.ConfigOverrides{})
+		c.factory = clientcmd.NewFactory(defaultCfg)
+	}
+	return c.factory, nil
+}
+
+// Clients returns clients for OpenShift and Kube
+func (c *ClientStartConfig) Clients() (*client.Client, *kclient.Client, error) {
+	f, err := c.Factory()
+	if err != nil {
+		return nil, nil, err
+	}
+	return f.Clients()
+}
+
+// OpenShiftHelper returns a helper object to work with OpenShift on the server
+func (c *ClientStartConfig) OpenShiftHelper() *openshift.Helper {
+	if c.openShiftHelper == nil {
+		c.openShiftHelper = openshift.NewHelper(c.dockerClient, c.HostHelper(), c.openShiftImage(), openShiftContainer, c.PublicHostname, c.RoutingSuffix)
+	}
+	return c.openShiftHelper
+}
+
+// HostHelper returns a helper object to check Host configuration
+func (c *ClientStartConfig) HostHelper() *host.HostHelper {
+	if c.hostHelper == nil {
+		c.hostHelper = host.NewHostHelper(c.dockerClient, c.openShiftImage(), c.HostVolumesDir)
+	}
+	return c.hostHelper
+}
+
+// DockerHelper returns a helper object to work with the Docker client
+func (c *ClientStartConfig) DockerHelper() *dockerhelper.Helper {
+	if c.dockerHelper == nil {
+		c.dockerHelper = dockerhelper.NewHelper(c.dockerClient)
+	}
+	return c.dockerHelper
+}
+
+func (c *ClientStartConfig) importObjects(out io.Writer, locations map[string]string) error {
+	f, err := c.Factory()
+	if err != nil {
+		return err
+	}
+	for name, location := range locations {
+		glog.V(2).Infof("Importing %s from %s", name, location)
+		err = openshift.ImportObjects(f, openShiftNamespace, location)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *ClientStartConfig) openShiftImage() string {
+	return fmt.Sprintf("openshift/origin:%s", c.Tag)
```

Comment:
This needs to be fully parametrizable, like the openshift start command.  Please open a follow up for it.